### DISCUSSION TITLE
No warning when project is build

### DISCUSSION THREAD

**Discussion by v-ghost03** [tbroyer/gradle-errorprone-plugin#61] [Q&A] [answered]
Hello, I have a problem. When I rebuild the project in `gradle`,  no warnings are displayed. The command `./gradle -clean` does not help me. I can see warnings only once time after cache cleaning or changing the `errorprone-plugin` version.

**Comment by tbroyer** [comment, marked as answer]
If I understand correctly, this is the expected behavior: without `-Werror`, warnings do not fail the build, so when rebuilding without code change the task is up-to-date, and warnings aren't displayed. If you use the build cache, the task result is taken from the cache, so again no compilation and no warnings.

This is Gradle's standard behavior for compilation warnings, that'd be the same for, e.g. deprecation warnings.
And with incremental recompilation, you'd also only have warnings for recompiled classes.

**Reply by v-ghost03** [reply]
Yes, you are right. I think using  `errorprone` without `-Werror` is possible when the `gradle` cache is off. I may turn off cache using command-line arguments for certain cases.  Is possible to create a feature in the next `errorprone` releases that allows managing to use a cache? I think it is would be more convenient.
Thanks for your reply!

**Reply by tbroyer** [reply]
> Is possible to create a feature in the next `errorprone` releases that allows managing to use a cache?

No, this is not the plugin's role (as I said, you'd have the same behavior with Xlint warnings, or warnings emitted by annotation processors, etc.)

You can do it yourself though by configuring the task outputs' `cacheIf`/`doNotCacheIf` and/or `upToDateWhen`. See https://blog.gradle.org/stop-rerunning-tests though, which explains why it's probably not a good idea (in other words, I'd personally stick with `cleanCompileJava` or `--rerun-tasks`, and `--no-build-cache`).

Because this is unrelated to ErrorProne or this plugin themselves (there are other warnings), and is directly related to incremental builds and build caching, I think you should bring this discussion to the Gradle forum to find the best solution (I, for one, always use `-Werror`, so I'm not affected).
See also https://github.com/gradle/gradle/issues/16984 (about Kotlin, but applies just as well to Java)

**Reply by v-ghost03** [reply]
Ok, Thanks
